db/active_replicator_pull.go

```diff
@@ -175,21 +176,64 @@ func (apr *ActivePullReplicator) reset() error {
 	return resetLocalCheckpoint(apr.config.ActiveDB, checkpointID)
 }
 
+type pullCheckpointerLookup struct {
+	m map[IDAndRev]string
+	l sync.Mutex
```

Comment:
Doing the id+rev -> seq mapping here adds a lock acquisition on every changes batch and rev/no rev message, in addition to the locking we're already doing in the checkpointer itself for AddExpectedSeq/AddProcessedSeq.  Given that, I'm wondering how much work it would be to push this handling down into the checkpointer (and so have the checkpointer work with id/rev/seq tuples) to avoid this extra lock.